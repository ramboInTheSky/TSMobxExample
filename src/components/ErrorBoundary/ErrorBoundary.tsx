import React from "react"

export class Boundary extends React.Component {
  state = { hasError: false }

  componentDidCatch(error: Error, info: any) {
    this.setState({ hasError: true })
    //   logErrorToMyService(error, info)
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong with this component</h1>
    }
    return this.props.children
  }
}

export default Boundary;
