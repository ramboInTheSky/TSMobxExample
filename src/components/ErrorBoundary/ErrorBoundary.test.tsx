import React from 'react'
import { shallow } from 'enzyme'
import { Boundary } from './ErrorBoundary'

describe('<Boundary />', () => {
	it('renders correctly', () => {
		expect(shallow(<Boundary />)).toMatchSnapshot()
	})

	// it('passes on props correctly', () => {
	// 	expect(shallow(<Boundary foo="bar">Hello</Boundary>)).toMatchSnapshot()
	// })

	// it('Does not re-render on children props change', () => {
	// 	const el: any = shallow(<Boundary foo="bar" />)
	// 	el.setProps({ children: 'Goodbye' })
	// 	el.update()
	// 	expect(el).toMatchSnapshot()
	// })
})
