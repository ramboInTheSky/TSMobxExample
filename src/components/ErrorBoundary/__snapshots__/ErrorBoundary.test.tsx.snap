// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Boundary /> renders correctly 1`] = `""`;
